src/main/kotlin/com/example/ajaxproject/controller/UserController.kt

```diff
@@ -44,4 +32,15 @@ class UserController @Autowired constructor(
         return ResponseEntity.noContent().build()
     }
 
+    @GetMapping("/{id}")
+    fun findUserById(@PathVariable id: Long): ResponseEntity<User?> {
+        val user = userService.findUserById(id)
+        return ResponseEntity.ok(user)
+    }
```

Comment:
we expect that user by id is exist. So, we can simplify the function:
```kotlin
    @GetMapping("/{id}")
    fun findUserById(@PathVariable id: Long): User = userService.findUserById(id)
``` 
Isn't it beautiful?

Note: you should have `@ExceptionHandler`(or @ControllerAdvice) that converts exception to 404.
More details:
- your service should throw. Let's say, custom UserNotFoundException.
- then an exception handler intercepts the exception and converts to 404 response